### PR TITLE
[Docs] Adds Deployment section to pull request template

### PR DESCRIPTION
**Pull request by mnigh**
## 👋 Introduction

This PR adds a Deployment section to the pull request template.

### CODE REVIEW

**Comment by vd1992** [comment]
Should deployment info go to the issue or the PR? 
The deployment tag gets added to the issue so I think it makes sense for deployment instructions to be added there too. I've added deployment info to PRs in the past though, I am not actually sure where it would be best placed

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -18,3 +18,7 @@ Assist reviewers with steps they can take to test that the PR does what it says
 ## 📸 Screenshot
 
 Add a screenshot (if possible).
+
+## 🚀 Deployment Notes
```
Comment (vd1992) [inline .github/pull_request_template.md:22]: Secondly, and of utmost importance 
Rocket... Or 
🚢 ship
:shipit: shipit
🛫 departure
🚚 truck
🖥️ ⌨️  computer and keyboard

**Comment by mnigh** [comment]
> Should deployment info go to the issue or the PR? The deployment tag gets added to the issue so I think it makes sense for deployment instructions to be added there too. I've added deployment info to PRs in the past though, I am not actually sure where it would be best placed

I put it in the pull request template rather than the issue templates based on this comment: 
>Team comment after moving to GitHub projects: deployment label still goes on issue, "Deployment Notes" section still goes in the PR. Thanks @petertgiles!

REF: https://github.com/GCTC-NTGC/gc-digital-talent/pull/5802#issuecomment-1450417763

**Comment by petertgiles** [comment]
Great idea!  I suggest filling it in a bit with more details like
- Examples of things needing deployment steps: Rerunning seeders, environment variable changes... others?
- Reminder to add deployment label to issue if steps needed
- Remove section if no steps needed

File: .github/pull_request_template.md (hunk 1 above)
Comment (petertgiles) [reply .github/pull_request_template.md:22]: truck ftw!

**Comment by petertgiles** [comment]
> I put it in the pull request template rather than the issue templates based on this comment:

My rationale: The deployment steps are related to the code you're merging, not the original issue description.  Conceptually I think it makes more sense to have the label on the PR as well but when you're viewing the sprint in Zenhub it only shows you the list of issues with their labels.  It's not easy for the deployer to list the PRs in the sprint so the label ends up on the issue.

File: .github/pull_request_template.md (hunk 1 above)
Comment (mnigh) [reply .github/pull_request_template.md]: 🚚, see 152a3b0a94ee80ac091b9841125802617d46f8e7.

**Review by petertgiles — APPROVED** [review]
:heart_eyes:

**Review by vd1992 — APPROVED** [review]
vroom vroom

**Pull request merged**